File: patterns/javascript-patterns/pattern31.js
// Pattern 31: ...

// * # # # # # #
// # # # # # * *
// * * * # # # #
// # # # * * * *
// * * * # # # #
// # # # # # * *
// * # # # # # #

const n = 7;
let x = 1;

for (let i = 1; i <= n; i++) {
    let row = '';

    for (let j = 1; j <= n; j++) {
        if (i % 2 === 1 && j <= x) {
            row += '* ';
        } else if (i % 2 === 0 && j >= n - x + 1) {
            row += '* ';
        } else {
            row += '# ';
        }
    }

    console.log(row);

    if (i <= n / 2) {
        x++;
    } else {
        x--;
    }
}

File: patterns/python-patterns/pattern31.py
# Pattern 31: ...

# * # # # # # #
# # # # # # * *
# * * * # # # #
# # # # * * * *
# * * * # # # #
# # # # # # * *
# * # # # # # #

n = 7
x = 1

for i in range(1, n + 1):
    for j in range(1, n + 1):
        if i % 2 == 1 and j <= x:
            print("* ", end="")
        elif i % 2 == 0 and j >= n - x + 1:
            print("* ", end="")
        else:
            print("# ", end="")
    print()
    if i <= n // 2:
        x += 1
    else:
        x -= 1
